File: MOanda/src/test/java/com/oanda/m/jmoanda/AccountTest.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package com.oanda.m.jmoanda;

import com.oanda.m.Client;
import junit.framework.TestCase;
import java.util.Map.Entry;
/**
 *
 * @author xavier
 */
public class AccountTest extends TestCase {
    
    public AccountTest(String testName) {
        super(testName);
    }

    @Override
    protected void setUp() throws Exception {
        super.setUp();
    }

    @Override
    protected void tearDown() throws Exception {
        super.tearDown();
    }

    // TODO add test methods here. The name must begin with 'test'. For example:
    public void testHello() throws Exception {
        Client client = new Client();
        client.login();
        client.home();
        System.out.println(client.summary().getBalance() + "");
        for (Entry<String, String> e : client.getAccounts().entrySet()) {
            System.out.println("id:" + e.getKey() + "value:" + e.getValue());
        }
        System.out.println(client.getAccount());
        client.setAccount("5835725");
        client.home();
        System.out.println(client.getAccount());
        System.out.println(client.summary().getBalance() + "");
        client.logout();
    }

}

File: FXHeatmapClient/src/test/scala/AccountTest.scala
import ch.olsen.fxheatmap.scala._

import junit.framework.TestCase

/**
 *
 * @author xavier
 */

class AccountScalaTest extends TestCase {
    


    // TODO add test methods here. The name must begin with 'test'. For example:
    def testHello() = {
        val client = Client("xtordoir@yahoo.co.uk","taxi2035")
        client.login
        
        client.addPair("EUR_USD");
        
        val hmd = client.fetch();
        System.out.println(hmd.get("EUR_USD", 4.5D));
        System.out.println(hmd.get("EUR_USD", 3.2D));
        System.out.println(hmd.get("EUR_USD", 1.6D));
        System.out.println(hmd.get("EUR_USD", 0.8D));
        System.out.println(hmd.get("EUR_USD", 0.4D));
        System.out.println(hmd.get("EUR_USD", 0.2D));
        System.out.println(hmd.get("EUR_USD", 0.1D));
        System.out.println(hmd.get("EUR_USD", 0.05D));
        System.out.println(hmd.get("EUR_USD", 0.02D));
        System.out.println(hmd.get("EUR_USD", 1.0));
    }

}